File: models/PersonalHistory.js
const PersonalHistorySchema = new mongoose.Schema({
    visitId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientVisit" },
    diet: {type: String,enums:["Vegetarian", "Non-Vegetarian"]},
    appetite: {type:String,enums:["Normal", "Reduced", "Increased"]},
    sleep: {type:String,enums:["Normal", "Reduced", "Increased"]},
    bladder: {type:String,enums:["Consistent", "Inconsistent"]},
    bowel: {type:String,enums:["Normal", "Loose-Motions", "Constipation"]},
    createdByUser:{type:mongoose.Schema.Types.ObjectId,ref:"User"},
  }, { timestamps: true });
  
  const PersonalHistory= mongoose.model("PersonalHistory", PersonalHistorySchema);
  export default PersonalHistory;

File: jstojv/src/main/java/com/samrat/jstojv/entity/PersonalHistory.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "personal_histories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PersonalHistory extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "visit_id")
    private PatientVisit visit;

    @Enumerated(EnumType.STRING)
    private Diet diet;

    @Enumerated(EnumType.STRING)
    private Appetite appetite;

    @Enumerated(EnumType.STRING)
    private Sleep sleep;

    @Enumerated(EnumType.STRING)
    private Bladder bladder;

    @Enumerated(EnumType.STRING)
    private Bowel bowel;

    @Column(name = "current_treatment")
    private String currentTreatment;

    private String comments;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdByUser;

    public enum Diet {
        Vegetarian, Non_Vegetarian
    }

    public enum Appetite {
        Normal, Reduced, Increased
    }

    public enum Sleep {
        Normal, Reduced, Increased
    }

    public enum Bladder {
        Consistent, Inconsistent
    }

    public enum Bowel {
        Normal, Loose_Motions, Constipation
    }
}
